RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -243,27 +227,22 @@ public Optional<ObjectNode> getDocument(String indexName, String documentId) {
     }
 
     /*
-     * Update a document using optimistic locking.  Returns an Optional; if the document was updated, it
-     * will be the new value and empty otherwise.
+     * Update a document using optimistic locking with the versioning info in an original copy of the doc that is passed in.
+     * Returns an Optional; if the document was updated, it will be the new value and empty otherwise.
      */
-    public Optional<ObjectNode> updateDocument(String indexName, String documentId, ObjectNode body) {
-        Optional<ObjectNode> document = getDocument(indexName, documentId);
-        if (document.isEmpty()) {
-            throw new UpdateFailed("Document not found: " + indexName + "/" + documentId);
-        }
-
+    public Optional<ObjectNode> updateDocument(String indexName, String documentId, ObjectNode newBody, ObjectNode originalCopy) {
```

Comment:
It looks like metadata is being pulled off the document, rather than extract this from a json data structure in the method lets specifically extract the seq id and pass it as a parameter since it change the behavior if it is present or not.  If present it fails to update if it doesn't match, if not present the document will be created.

Reply:
Created issue for follow up, closing out this thread and related
- https://github.com/opensearch-project/opensearch-migrations/issues/697